### DISCUSSION TITLE
How can I create dropdown with submenus in Vanilla JS with Floating-UI?

### DISCUSSION THREAD

**Discussion by MichaelRydl** [floating-ui/floating-ui#2344] [Q&A] [answered]
Hello everyone, 

I´d like to know how can I create the dropdown component with submenus in Vanilla JS with the Floating-UI? I found the example for React, but I think that´s not really applicable for the Vanilla JS component. 

When I try it by myself, I will up end with this wrongly positioned third-level submenu:
![image](https://github.com/floating-ui/floating-ui/assets/29844074/cd4c0e0f-ea14-4cde-98fb-4048afe69321)

Could you please someone help me and provide some example?

Thanks a lot.

**Comment by thetutlage** [comment]
Looking for same. I really like the low level APIs of floating UI and wondering if there can be some examples of creating list menu or context menu in VanillaJS

**Comment by FezVrasta** [comment]
Would the floating tree work for you? https://floating-ui.com/docs/FloatingTree

**Reply by MichaelRydl** [reply]
If I'm not mistaken the FloatingTree is only for React components, not for the case I need - Vanilla JS component.

**Comment by atomiks** [comment, marked as answer]
This library only supports anchor positioning for vanilla JS (which is useful by itself, but only one part of the story of a dropdown menu component). All interactions and other features are React-only. Making a properly functioning and accessible dropdown menu with submenus isn't an easy task. I'm not sure why the positioning is incorrect in your example because there's no code, but to make it fully accessible requires a LOT of work. 

Things like `safePolygon`, list navigation, typeahead, portaling and getting it to work with screen readers (especially iOS touch screen reader) are complex. Doing so without React's declarative rendering model is a huge pain, and to me, not worth attempting at all. It legitimately takes weeks and weeks of work to do it properly. I can only suggest you find a dropdown menu component for vanilla JS that tries to handle all of that, and only use the anchor positioning feature for it if it doesn't already offer that.
